getsentry/sentry
# Fix Polling Sorting on Stream

#### What's this PR do?

This PR fixes the sorting of new items added to the `Stream` from `CursorPoller`. Previously, new items were added in reverse order. Now they're not.

This PR also adds infrastructure that enables stubbing React components.

This fixes "Ensure polling is correctly sorting results" on issue #1585.
#### Where should the reviewer start?

`src/sentry/static/sentry/app/utils/streamManager.jsx`
#### How should this be manually tested?

Run `make test-js`

Alternatively, you can fake data coming back from the poll and watch it appear in the correct order in the Stream.
#### Any background context you want to provide?

The previous `StreamManager.unshift()` would reverse incoming arrays in an attempt to put the most recent items towards the beginning of the array. The polling call returns data in correctly sorted order, which should not be reversed.

We removed the call to `reverse` and wrote an additional test to highlight the expected behavior.
#### What gif best describes this PR or how it makes you feel?

![highfive](http://media.giphy.com/media/120jXUxrHF5QJ2/giphy.gif)
